lib/src/services/auth_service.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:convert';
+import 'package:basic_authentication_flutter_challenge/src/data/constants/api_constants.dart';
+import 'package:basic_authentication_flutter_challenge/src/data/helper_classes/http_client.dart';
+import 'package:basic_authentication_flutter_challenge/src/services/current_auth_user.dart';
+import 'package:basic_authentication_flutter_challenge/src/services/tokens_service.dart';
+import 'package:flutter/foundation.dart';
+
+class AuthService extends ChangeNotifier {
+  final httpClient = const HttpClient();
+  final TokensService tokens;
+  final CurrenAuthtUser authUser;
+
+  AuthService({required this.tokens, required this.authUser});
+
+  Future<bool> isAuthenticated() async {
+    final token = await tokens.getLocalAccessToken();
+    return token.isNotEmpty;
+  }
+
+  Future<void> login(String username, String password) async {
+    // Do basic auth
+    String base64Encoded = base64.encode(utf8.encode('$username:$password'));
+    await tokens.updateAccessTokenLocally('Basic $base64Encoded');
```

Comment:
It seems you are storing the encoded value in local storage then we override the value with the valid token if received.

I would recommend to separate the values, and each one has its own local key.

Reply:
Separation has no advantage here, because you will never need both tokens at the same time.

Reply:
For code scalability, and readability you should separate them.